src/Cool/DidYouKnow/didYouKnowModal.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <vector>
+#include "cereal/types/chrono.hpp"
+class DidYouKnowModal {
+public:
+    void open();
+
+    void open_ifn();
+
+    void imgui_window();
+
+private:
+    bool _has_been_opened = false;
+    bool _show_all_tips   = false;
+
+    size_t _current_tip_index = 0;
+
+    std::chrono::steady_clock::time_point _timestamp_last_opening;
+
+    void prepare_next_tip();
```

Comment:
Our convention is to put private methods in a separate `private:` section, and to put them before the private variables